### PR TITLE
setuptools: Bump version to 54.0.0

### PR DESCRIPTION
**Pull request by phlax**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Does not include precompiled binaries, eg. `.par` files. See [CONTRIBUTING.md](/CONTRIBUTING.md) for info
- [?] Tests for the changes have been added (for bug fixes / features)
- [?] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [X] Other... Please describe:

dependency update

## What is the current behavior?

rules_python uses old version of setuptools which can break some packages on windows

Issue Number: #421 

## What is the new behavior?

rules_python uses current setuptools stable (54.0) which fixes above problem

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

?

### CODE REVIEW

File: python/requirements.txt
Comment (phlax) [inline python/requirements.txt]: not sure why this wont let me set newer setuptools - versions available only seem to go to 44.1

Comment (aaliddell) [reply python/requirements.txt]: Version 44 was the last to support both Python 2 & 3. So until rules_python fully drops Python 2 support, it may be stuck with this version.

File: python/pip_install/repositories.bzl
Comment (phlax) [inline python/pip_install/repositories.bzl:19]: @aaliddell 

> Version 44 was the last to support both Python 2 & 3. So until rules_python fully drops Python 2 support, it may be stuck with this version.

does that also preclude this change ?

Comment (aaliddell) [reply python/pip_install/repositories.bzl:19]: Perhaps, but I think the new `pip_install` is only intended to support Python 3 (#409). The issue seems to be that there's also the older packaging rules still present, which support Python 2.

Comment (phlax) [reply python/pip_install/repositories.bzl:19]: it would be really helpful if this were included from our pov, but my bazel knowledge is a bit limited so im not totally sure of the effect

**Comment by groodt** [comment]
I personally think this will work itself out and be easier to manage once the other train of PRs is merged as mentioned here https://github.com/bazelbuild/rules_python/issues/421#issuecomment-788350959

In summary:
* Maintainers to cut new release that will be the last release with any semblance of Python 2.7 and 3.5 support for pip_install (it was never really supported, but kinda worked sometimes).
* Remove Python 3.5 support from CI #417
* Upgrade to pip 21.0.1 (pip dropping Python 2.7 and this adds support for Big Sur and Apple Silicon wheels. The setuptools upgrade could possibly be added here tbh, since setuptools > 45 requires Python > 3.5. https://github.com/bazelbuild/rules_python/pull/412

This change can't really be done until the above is done. It might also be possible to include this as part of #412, but perhaps also as a fast-follow too, since it is an unrelated changed to #412.

**Review by hrfuller — APPROVED** [review]

**Pull request merged**
